python/lsst/pipe/base/_instrument.py

```diff
@@ -735,3 +744,27 @@ def _make_default_dimension_packer(
             IDs into integers.
         """
         return config_dict.apply_with(default, data_id, is_exposure=is_exposure)
+
+    @classmethod
+    def group_name_to_group_id(cls, group_name: str) -> int:
```

Comment:
@TallJimbo you think this should move to `lsst.obs.base.Instrument` rather than being in the base class?

Reply:
One downside of moving this to obs_base is that mypy is clever enough to notice that `Instrument.fromName` is not returning something that has that method. I've changed the return types to `Self` on `Instrument` for the constructors but that's not strictly true in that those methods are not going to magically return the obs_base variant if you call `lsst.obs.base.Instrument.fromName("name.that.inherits.from.pipe.base.Instrument")`.

Reply:
I've modified the pipe_base variant to use class methods and then check that the returned instrument instance is a subclass of the class you used to call `fromName` et al. That will at least let things crash early if people are using obs_base define visits without using an obs_base Instrument.